client/components/data/query-sites/index.jsx

```diff
@@ -17,66 +13,74 @@ import { requestSites, requestSite } from 'state/sites/actions';
 import { getPreference } from 'state/preferences/selectors';
 import getPrimarySiteId from 'state/selectors/get-primary-site-id';
 
-class QuerySites extends Component {
-	UNSAFE_componentWillMount() {
-		this.requestAll( this.props );
-		this.requestPrimary( this.props );
-		this.requestRecent( this.props );
-		this.requestSingle( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.siteId !== this.props.siteId ) {
-			this.requestSingle( nextProps );
-		}
+const getRecentSites = state => getPreference( state, 'recentSites' );
 
-		if ( nextProps.primaryAndRecent && nextProps.primarySiteId !== this.props.primarySiteId ) {
-			this.requestPrimary( nextProps );
-		}
+const requestAll = () => ( dispatch, getState ) => {
+	if ( ! isRequestingSites( getState() ) ) {
+		dispatch( requestSites() );
+	}
+};
 
-		if (
-			nextProps.primaryAndRecent &&
-			! isEqual( nextProps.recentSiteIds, this.props.recentSiteIds )
-		) {
-			this.requestRecent( nextProps );
-		}
+const requestSingle = siteId => ( dispatch, getState ) => {
+	if ( siteId && ! isRequestingSite( getState(), siteId ) ) {
+		dispatch( requestSite( siteId ) );
 	}
+};
 
-	requestAll( props ) {
-		if ( props.allSites && ! props.requestingSites ) {
-			props.requestSites();
-		}
+const requestPrimary = siteId => ( dispatch, getState ) => {
+	if ( siteId && ! hasAllSitesList( getState() ) && ! isRequestingSite( getState(), siteId ) ) {
+		dispatch( requestSite( siteId ) );
 	}
+};
 
-	requestPrimary( props ) {
-		if ( props.primaryAndRecent && ! props.hasAllSitesList ) {
-			const { primarySiteId, isRequestingPrimarySite } = props;
+const requestRecent = siteIds => ( dispatch, getState ) => {
+	if ( siteIds.length && ! hasAllSitesList( getState() ) ) {
+		const isRequestingSomeSite = siteIds.some( siteId => isRequestingSite( getState(), siteId ) );
 
-			if ( primarySiteId && ! isRequestingPrimarySite ) {
-				props.requestSite( primarySiteId );
-			}
+		if ( ! isRequestingSomeSite ) {
```

Comment:
I like to write conditions like this as a series of early returns:
```js
if ( ! preConditionA ) {
  return;
}

if ( ! preconditionB ) {
  return;
}

performTheAction();
```
It keeps nesting minimal, clearly states what conditions need to be satisfied and makes it more clear what is the main task performed.

Reply:
Done.